docs/ADR/0053-double-colon-type-annotation-syntax.md

```
@@ -0,0 +1,338 @@
+  includes: element :: Object -> Boolean => ...
+  inject: initial into: block :: Block => @primitive "inject:into:"
+  collect: block :: Block -> Self => ...
+  detect: block :: Block -> Object | Nil => ...
+  detect: block :: Block ifNone: noneBlock :: Block -> Object => ...
+```
+
+The selector `detect:ifNone:` is now immediately distinguishable from the parameter annotations `block :: Block` and `noneBlock :: Block`.
+
+### Error Examples
+
+```
```

markdownlint-cli2 (0.21.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)